### PR TITLE
Added pictures to flashcards, a few misc image changes

### PR DESCRIPTION
**Pull request by Omicron3141**
Also added default images, resized flashcard images

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -56,11 +56,24 @@ def destroy
...
         @course = @user.courses.find(params[:id])
+        # randomly select one student from the course roster
         @student = @course.students.sample
-        render(json:  @student.to_json)
+        # if there is a student to show
+        if @student != nil
+            # convert the student object to a hash
+            @student_hash = @student.attributes
+            # add the student's portrait url to the hash
+            @student_hash[:portrait_url] = @student.portrait.url(:flashcard)
```
Comment (kcwoolverton) [inline app/controllers/courses_controller.rb:70]: Out of curiosity, why do you need this line? Is the portrait_url not included in the student attributes?

Comment (Omicron3141) [reply app/controllers/courses_controller.rb:70]: For some reason no. It's an attachment, so I suspect it comes in from another database table.

Comment (kcwoolverton) [reply app/controllers/courses_controller.rb:70]: Hmmm... could this cause problems down the line, or is it just a quirk?

File: app/views/courses/show.html.erb [hunk 2]
```diff
@@ -1,21 +1,32 @@
 <script>
+// called when the "draw flashcard" button is pressed
 function drawFlashcard() {
+  // create a new AJAX request object
   var xhttp = new XMLHttpRequest();
+  // tell the request object that when it's state changes, it should check if it good a good reply back, and then call populateFlashcard
```
Comment (kcwoolverton) [inline app/views/courses/show.html.erb]: it's --> its

File: app/controllers/courses_controller.rb (hunk 1 above)
Comment (Omicron3141) [reply app/controllers/courses_controller.rb:70]: I think it's better this way, we don't want image changes to affect the student table. Also, I think it's paperclip, so we might not be able to change it.

**Review by kcwoolverton — APPROVED** [review]

**Pull request merged**